bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -388,44 +375,45 @@ protected Channel createNettyChannel(final String host) throws IOException {
+    if (managedChannels.isEmpty()) {
+      return;
     }
-    if (!channelImpl.isTerminated()) {
-      // Sometimes, gRPC channels don't close properly. We cannot explain why that happens,
-      // nor can we reproduce the problem reliably. However, that doesn't actually cause
-      // problems. Synchronous RPCs will throw exceptions right away. Buffered Mutator based
-      // async operations are already logged. Direct async operations may have some trouble,
-      // but users should not currently be using them directly.
-      LOG.trace("Could not close the channel after %d ms.", timeoutMs);
+    long timeoutNanos = TimeUnit.SECONDS.toNanos(10);
+    long endTimeNanos = System.nanoTime() + timeoutNanos;
+    for (ManagedChannel channel : managedChannels) {
+      channel.shutdown();
     }
-  }
-
-  @Override
-  public void close() throws IOException {
-    List<ListenableFuture<Void>> closingChannelsFutures = new ArrayList<>();
-    ListeningExecutorService listenableBatchPool = MoreExecutors
-        .listeningDecorator(BigtableSessionSharedThreadPools.getInstance().getBatchThreadPool());
-    for (final Closeable clientCloseHandler : clientCloseHandlers) {
-      closingChannelsFutures.add(listenableBatchPool.submit(createCallable(clientCloseHandler)));
+    for (ManagedChannel channel : managedChannels) {
+      long awaitTime = endTimeNanos - System.nanoTime();
+      if (awaitTime < 0) {
+        break;
+      }
+      try {
+        channel.awaitTermination(awaitTime, TimeUnit.NANOSECONDS);
+      } catch (InterruptedException e) {
+        throw new IOException("Interrupted while closing the channelPools");
+      }
     }
-    get(Futures.allAsList(closingChannelsFutures), "waiting for channels to be closed");
-  }
-
-  protected Callable<Void> createCallable(final Closeable clientCloseHandler) {
-    return new Callable<Void>() {
-      @Override
-      public Void call() throws Exception {
-        clientCloseHandler.close();
-        return null;
+    for (ManagedChannel channel : managedChannels) {
+      if (!channel.isTerminated()) {
+        // Sometimes, gRPC channels don't close properly. We cannot explain why that happens,
+        // nor can we reproduce the problem reliably. However, that doesn't actually cause
+        // problems. Synchronous RPCs will throw exceptions right away. Buffered Mutator based
+        // async operations are already logged. Direct async operations may have some trouble,
+        // but users should not currently be using them directly.
+        //
+        // NOTE: We haven't seen this problem since removing the RefreshingChannel
+        LOG.info("Could not close the channel after 10 seconds.");
+        break;
       }
-    };
+    }
+    for (Iterator<ManagedChannel> iterator = managedChannels.iterator(); iterator.hasNext();) {
+      ManagedChannel channel = iterator.next();
+      if (channel.isTerminated()) {
+        iterator.remove();
```

Comment:
Could we merge both for loops? What is the point of removing managedChannels one by one? could we empty it at the end? why leave the not terminated ones? they wont be used by anyone later on.